src/lib/network_pool/snark_pool.ml

```diff
@@ -148,19 +165,33 @@ struct
                :: acc ))
 
       let all_completed_work (t : t) : Transaction_snark_work.Info.t list =
-        Statement_table.fold ~init:[] t.snark_table
+        Statement_table.fold ~init:[] t.snark_tables.all
           ~f:(fun ~key ~data:{proof= _; fee= {fee; prover}} acc ->
             let work_ids = Transaction_snark_work.Statement.work_ids key in
             {Transaction_snark_work.Info.statements= key; work_ids; fee; prover}
             :: acc )
 
+      (** True when there is no active transition_frontier or
+          when the refcount for the given work is 0 *)
```

Comment:
Just noticing that this comment seems wrong - it doesn't check that the refcount is zero?